internal/core/data/router.go

```diff
@@ -72,3 +80,212 @@ func LoadRestRoutes() *mux.Router {
+	id, err := url.QueryUnescape(vars["deviceId"])
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Problem unescaping URL: " + err.Error())
+		return
+	}
+
+	switch r.Method {
+	case http.MethodGet:
+		// Check device
+		count, err := countByDevice(id)
+		if err != nil {
+			LoggingClient.Error(fmt.Sprintf("error checking device %s %v", id, err))
+			switch err := err.(type) {
+			case types.ErrNotFound:
+				http.Error(w, err.Error(), http.StatusNotFound)
+				return
+			default: //return an error on everything else.
+				http.Error(w, err.Error(), http.StatusServiceUnavailable)
+				return
+			}
+		}
+
+		// Return result
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(strconv.Itoa(count)))
+		break
+	}
+}
+
+// Remove all the old events and associated readings (by age)
+// event/removeold/age/{age}
+func eventByAgeHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	vars := mux.Vars(r)
+	age, err := strconv.ParseInt(vars["age"], 10, 64)
+
+	// Problem converting age
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Error converting the age to an integer")
+		return
+	}
+
+	switch r.Method {
+	case http.MethodDelete:
+		LoggingClient.Info("Deleting events by age: " + vars["age"])
+
+		count, err := deleteByAge(age)
+		if err != nil {
+			LoggingClient.Error(err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		// Return the count
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(strconv.Itoa(count)))
+	}
+}
+
+/*
+Handler for the event API
+Status code 404 - event not found
+Status code 413 - number of events exceeds limit
+Status code 503 - unanticipated issues
+api/v1/event
+*/
+func eventHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Body != nil {
+		defer r.Body.Close()
+	}
+
+	switch r.Method {
+	// Get all events
+	case http.MethodGet:
+		events, err := getEvents(configuration.ReadMaxLimit)
+		if err != nil {
+			LoggingClient.Error(err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		encode(events, w)
+		break
+		// Post a new event
+	case http.MethodPost:
+		var e models.Event
+		dec := json.NewDecoder(r.Body)
+		err := dec.Decode(&e)
+
+		// Problem Decoding Event
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			LoggingClient.Error("Error decoding event: " + err.Error())
+			return
+		}
+
+		LoggingClient.Info("Posting Event: " + e.String())
+
+		newId, err := addNew(e)
+		if err != nil {
+			switch t := err.(type) {
+			case errors.ErrValueDescriptorNotFound:
+			case errors.ErrValueDescriptorInvalid:
+				http.Error(w, t.Error(), http.StatusBadRequest)
+			default:
+				http.Error(w, t.Error(), http.StatusInternalServerError)
+			}
+			LoggingClient.Error(err.Error())
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(newId))
```

Comment:
What's the `Content-Type` here?

Reply:
@tsconn23 do you plan to set `Content-Type` header there?

Reply:
@drasko It is currently defaulting to text/plain. There is a blackbox test that expects this value and it's passing. If I set it explicitly to "text/plain", it will be different from the other content types in the same file so you have to promise not to call that out. If I leave it as is, since it's working, then we can address as part of the later effort to ensure all requests return valid JSON.